test/src/unittests/filters/test_medianfilter.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+
+from essentia_test import *
+from math import *
+from numpy import *
+
+from essentia import *
+from math import *
+import essentia.standard as std
+
+
+class TestMedianFilter(TestCase):
+
+    def testRegressionRange(self):
+        sr = 44100.
+        pi2 = 2*pi
+        signal = [.25*cos(t*pi2*5/sr) + \
+                  .25*cos(t*pi2*50/sr) + \
+                  .25*cos(t*pi2*500./sr) + \
+                  .25*cos(t*pi2*5000./sr)
+                  for t in range(44100)]
+
+        filteredSignal = MedianFilter(kernelSize=11)(signal)
+
+        s = Spectrum()(signal)
+        sf = Spectrum()(filteredSignal)
+
+        for i in range(1000):
+            if s[i] > 10:
+                self.assertTrue(sf[i] > 0.5*s[i])
+
+        for i in range(1001, len(sf)):
+            if s[i] > 10:
+                self.assertTrue(sf[i] < 0.5*s[i])
+    
+    def testRegressionPeaks(self):
+        sr = 44100
+        index=0
+        clean_signal = []
+        # The "unpython" While loop used because this code sample was tested like this on a 
+        # Jupyter notebook for plotting.
+        # The above "...for t...range()" threw errors in that environment.
+        while index < 1000:
+            original_signal_pt = .25 * cos((index/sr)  * 5 * 2*pi) \
+                                +.25 * cos((index/sr)  * 50 * 2*pi) \
+                                +.25 * cos((index/sr)  * 500 * 2*pi) \
+                                +.25 * cos((index/sr)  * 5000 * 2*pi)
+            clean_signal.append(original_signal_pt)
+            index+=1
+
+        medianfilteredSignal = std.MedianFilter()(clean_signal)
+        smf = std.Spectrum()(medianfilteredSignal)
+        self.assertAlmostEqual(smf[11], 97.21687, 8)      
+        self.assertAlmostEqual(smf[113], 35.783447, 8)      
```

Comment:
Here the same concept as for MaxFilter: use a manual test instead of relying on a regression re-run on a complex input. At each position *i* the expected output is a median of the values *i-1*, *i*, and *i+1* for the filter of size 3. The beginning and the end [are padded](https://github.com/MTG/essentia/blob/master/src/algorithms/filters/medianfilter.cpp#L57) with the first and last values in the input vector.

Reply:
testRegression was rewritten using the same principle as in maxFilter.
A input vector whose values can be manually determined . It was done for kernel sizes 3 and 4.
testRegressionRange was left as is. 
(the idea for this test came from the low pass filter test that works with config param. "cut off freq." instead of kernel size)